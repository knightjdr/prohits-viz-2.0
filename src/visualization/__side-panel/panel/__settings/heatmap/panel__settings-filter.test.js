import React from 'react';
import { shallow } from 'enzyme';

import Filter from './panel__settings-filter';

const changeSetting = jest.fn();
const updateSetting = jest.fn();

const settings = {
  abundanceCap: 50,
  minAbundance: 0,
  primaryFilter: 0.01,
  secondaryFilter: 0.05,
};

describe('Filter panel settings', () => {
  let wrapper;

  beforeAll(() => {
    wrapper = shallow(
      <Filter
        changeSetting={changeSetting}
        settings={settings}
        storeSettings={settings}
        updateSetting={updateSetting}
      />,
    );
  });


  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call changeSetting from first input', () => {
    changeSetting.mockClear();
    wrapper.find('Input').at(0).simulate('change');
    expect(changeSetting).toHaveBeenCalled();
  });

  it('should call updateSetting from first input', () => {
    updateSetting.mockClear();
    wrapper.find('Input').at(0).simulate('click');
    expect(updateSetting).toHaveBeenCalled();
  });

  it('should call changeSetting from second input', () => {
    changeSetting.mockClear();
    wrapper.find('Input').at(1).simulate('change');
    expect(changeSetting).toHaveBeenCalled();
  });

  it('should call updateSetting from second input', () => {
    updateSetting.mockClear();
    wrapper.find('Input').at(1).simulate('click');
    expect(updateSetting).toHaveBeenCalled();
  });

  it('should call changeSetting from third input', () => {
    changeSetting.mockClear();
    wrapper.find('Input').at(2).simulate('change');
    expect(changeSetting).toHaveBeenCalled();
  });

  it('should call updateSetting from third input', () => {
    updateSetting.mockClear();
    wrapper.find('Input').at(2).simulate('click');
    expect(updateSetting).toHaveBeenCalled();
  });

  it('should call changeSetting from fourth input', () => {
    changeSetting.mockClear();
    wrapper.find('Input').at(3).simulate('change');
    expect(changeSetting).toHaveBeenCalled();
  });

  it('should call updateSetting from fourth input', () => {
    updateSetting.mockClear();
    wrapper.find('Input').at(3).simulate('click');
    expect(updateSetting).toHaveBeenCalled();
  });
});
