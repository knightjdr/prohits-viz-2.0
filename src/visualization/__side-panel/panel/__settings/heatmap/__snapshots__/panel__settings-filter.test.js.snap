// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter panel settings should match snapshot 1`] = `
<div
  className="panel__settings-section"
>
  <Input
    field="abundanceCap"
    name="Abundance cap"
    onChange={[Function]}
    onClick={[Function]}
    store={50}
    value={50}
  />
  <Input
    field="minAbundance"
    name="Abundance minimum"
    onChange={[Function]}
    onClick={[Function]}
    store={0}
    value={0}
  />
  <Input
    field="primaryFilter"
    name="Primary filter"
    onChange={[Function]}
    onClick={[Function]}
    step={0.01}
    store={0.01}
    value={0.01}
  />
  <Input
    field="secondaryFilter"
    name="Secondary filter"
    onChange={[Function]}
    onClick={[Function]}
    step={0.01}
    store={0.05}
    value={0.05}
  />
</div>
`;
